src/worker.js

```diff
@@ -14,7 +14,7 @@ var Module = {};
 
 #if ENVIRONMENT_MAY_BE_NODE
 // Node.js support
-var ENVIRONMENT_IS_NODE = typeof process == 'object' && typeof process.versions == 'object' && typeof process.versions.node == 'string';
+var ENVIRONMENT_IS_NODE = typeof Deno == void 0 && typeof process == 'object' && typeof process.versions == 'object' && typeof process.versions.node == 'string';
```

Comment:
So the goal here is not to set ENVIRONMENT_IS_NODE when running on deno?

How much will actually work in that case?   Do we need `ENVIRONMENT_MAY_BE_DENO` to get things to actually work correctly, or does some stuff just work as long as we don't set `ENVIRONMENT_IS_NODE`.   

What `ENVIRONMENT_IS_XXX` variable would be set in that case?  I guess none of them?   I'd be somewhat surprised if that works.

Reply:
It’s currently working fine on Deno. But sometimes the `process` object is made available in the Deno environment for compatibility with Node.js.

Reply:
How you would feel about an opt-in `-sENVIRONMENT=deno` flag?  

Even if we move forward without that, it seems like maybe we should add some basic tests if we are going to claim to run under deno.

Reply:
I think it’d be cool. It would just need to remove the “is Node.js” checks and think as if it is a browser and everything will work fine.

Reply:
So we should set `ENVRIONMENT_IS_WEB` under deno you think?   Does that already get set today?

Reply:
Wait, do you already have an option to not include the Node.js checks in the generated .js?

Reply:
Yes you can do `-sENVRIONMENT=web` if you only want to target the web (for example).   (The default is to target node+web)

Reply:
Well, I’ll try that and close this if it worked. Thanks!

Reply:
I just tried `ENVIRONMENT=web` and it worked. But I still think this change makes sense if we want to target the three at once. And I want to do that now in a lib of mine.

Reply:
Do threads work under Deno via the Web's Worker API?    (if not then I don't think we need to modify this version of the ENVIRONMENT_IS_NODE).

Reply:
Also, are you saying the deno defines `process.versions.node` ?  That seems rather odd.

Reply:
Yes, threads work under Deno via Web Worker API. No, Deno doesn’t define `process.versions.node`, rather as I said sometimes its environment is modified for compatibility with Node.js modules, for example by [esm.sh](https://esm.sh).

Reply:
I think trying to play cat a mouse with folks that are trying to polyfill node by defining `process.versions.node` inside of something that isn't not be the right direction.    If esm.sh is trying to be compatible with node by declaring `process.versions.node` is seems reasonable to take them at their word... otherwise we basically trying to tell the difference between real node and pollyfilled/fake node.   

On the other hand I'm not an expert in this realm so maybe this kind of thing is normal?   Do you know of the best/recommend way to detect "real" node?